### ISSUE TITLE
Improve importing and exporting capability

### ISSUE THREAD

**Issue by xplosionmind** [omnivore-app/omnivore#2149]
So far, except for Matter, [the only way to import stuff is via the API](https://docs.omnivore.app/using/importing.html).

It would be extremely useful if importing and exporting functionality could be added within the (web)app.

Furthermore, not only a mere list of links should be imported, but also:

- tags
- location (Archive, Inbox, favorites)
- date added
- date last opened
- …?

Personally, I am interested in importing from [Wallabag](https://wallabag.org).

**Comment by jacksonh** [comment]
I think what we can do is start to develop a standard CSV format, and then write adapters to get things like Wallabag and Matter into that format.

**Comment by thiswillbeyourgithub** [comment]
Fyi related to #2094

**Comment by sopoforic** [comment]
I want to note something that I've just realized, in connection with this: if you use the email feature of Omnivore to subscribe to newsletters, then you cannot even rely on the wayback machine + highlight export in case omnivore vanishes, and you won't have the newsletters in your own email, either, so this is a bit more serious for those cases.

**Comment by jacksonh** [comment]
> I want to note something that I've just realized, in connection with this: if you use the email feature of Omnivore to subscribe to newsletters, then you cannot even rely on the wayback machine + highlight export in case omnivore vanishes, and you won't have the newsletters in your own email, either, so this is a bit more serious for those cases.

Yeah when we build an exporter tool we definitely wont make it URL only, it should export the full content (original + readable). You can do this now with the api-demo app, it just all needs to be packaged up into a standalone application.

**Comment by watsonbox** [comment]
Indeed, I discovered Omnivore myself recently, and it's surprisingly close to being a perfect read-it-later solution, save for the lack of export. I've started with the free hosted service which is great, but plan to self-host in future. Okay, I could put together a script to move data between the instances using the API, but it's a shame that's not available out of the box.

This is a matter of data _portability_. In the unlikely event of Omnivore disappearing all of a sudden, I'm satisfied that the Obsidian sync feature (also wonderful) provides a good solution to data _backup_ should it ever get to that. That's just my use-case of course, as not all users will be able to take advantage of the notes sync facility.

**Comment by Cito** [comment]
I have provided a simple export script here: https://github.com/Cito/omnivore-export

**Comment by whatfuturedotlol** [comment]
hey, i'd also love a raw export of everything. my current flow backs up all the original content in a dropbox and also on my nas. @Cito am i correct in thinking your script only backs up the links, not the actual content itself?

**Comment by Cito** [comment]
> your script only backs up the links, not the actual content itself

There is now a flag that also allows to export the content.

**Comment by candideu** [comment]
Any news on the export function?

**Comment by Podginator** [comment]
Closing due to age, and the self-hosted nature of Omnivore. Done as a batch action. If this issue still exists, please reopen it.
